### PR TITLE
feat: reth node is starting, handling one event, block height is impl…

### LINKED ISSUE #2: getBlockHeight: current height of SideCar node
### Method: `getBlockHeight`

**Description**

The `getBlockHeight` API call will return the block height that SideCar is currently synced to, considering only finalized L1 blocks.

**Request**

```json
{
  "jsonrpc": "2.0",
  "method": "getBlockHeight",
  "params": [],
  "id": 1
}
```
### Explanation:
- **jsonrpc**: Specifies the version of the JSON-RPC protocol, which is `2.0` in this case.
- **method**: The name of the method being called, which is `"getBlockHeight"`.
- **params**: An empty array `[]` because the `getBlockHeight` method does not require any parameters.
- **id**: A unique identifier for the request, which can be any number or string. Here, it is set to `1`.

**Request Parameters:**

none

**Response**

```json
{
  "jsonrpc": "2.0",
  "result": "538458",
  "id": 1
}
```

- **jsonrpc**: Version of the JSON-RPC protocol.
- **result**: The block height, returned as a decimal string (e.g., `"538458"`).
- **id**: The same `id` as sent in the request, which is `1` in this case.

### PR DESCRIPTION
**Pull request by jac18281828**
closes #2 

This pr introduces Reth.   Node is starting and a simple function "getBlockHeight" is implemented, documented and tested.

try me:
```bash
curl -X POST http://127.0.0.1:8545 \
        -H -d '{ "jsonrpc": "2.0", "method": "layr_getBlockHeight", "id": 1 }' 
```

Result
```json
{"jsonrpc":"2.0","id":1,"result":0}
```

I still need to setup a testnet and confirm the event handling.

### CODE REVIEW

File: layr_rpc/src/lib.rs [hunk 1]
```diff
@@ -1,30 +1,119 @@
...
+        Self {
+            block_data: block_data.clone(),
+        }
+    }
+}
 
 #[async_trait]
-impl RpcServer for RpcServerImpl {
+impl LayrRpcServer for LayrRpcServerImpl {
     async fn get_block_height(&self) -> Result<usize, ErrorObjectOwned> {
```
Comment (ypatil12) [inline layr_rpc/src/lib.rs:113]: We can get this directly from Reth? Or is the purpose here just to demonstrate exex writing the block_height to an external process?

File: layr_rpc/src/lib.rs [hunk 2]
```diff
@@ -1,30 +1,119 @@
...
     #[method(name = "healthCheck")]
     async fn health_check(&self) -> Result<String, ErrorObjectOwned>;
 }
 
-pub struct RpcServerImpl;
+pub struct LayrRpcServerImpl {
+    block_data: Arc<RwLock<Block>>,
+}
+
+impl LayrRpcServerImpl {
```
Comment (ypatil12) [inline layr_rpc/src/lib.rs:103]: Nit: we may want to double check on usage of Layr. We've deprecated its usage everywhere (except the repo itself lol)

File: layr_rpc/src/lib.rs (hunk 1 above)
Comment (ypatil12) [reply layr_rpc/src/lib.rs:113]: ie. `eth_blockNumber`. Do we also want to support JSON-RPC API with the `layr_rpc`?

File: sidecar/src/main.rs [hunk 3]
```diff
@@ -1,45 +1,188 @@
...
+//!     E
+//!     F
+//!     end
+//! ```
 //!
+//! SideCar, developed by Eigen Labs, is an open-source, permissionless, verified indexer that enables AVS co-processors or roll-ups to access the state of the Eigen Layer Ethereum protocol in real-time. This tool is designed to provide comprehensive and real-time protocol data access to various stakeholders within the Eigen Layer ecosystem.
+//!
+//! ### Purpose and Benefits
+//!
+//! SideCar addresses the high demand for protocol data by offering a definitive, self-service, and permissionless view of the Eigen Layer. It provides real-time views of delegated and slashable stakes, full visibility into AVS payment schedules, direct access to AVS and Operator metadata, and the ability to generate proofs for rewards.
```
Comment (ypatil12) [inline sidecar/src/main.rs:34]: nit: payment -> reward

File: sidecar/src/main.rs [hunk 4]
```diff
@@ -1,45 +1,188 @@
...
+//!
+//! ### Key Features
+//!
+//! - **Docker-based Deployment**: SideCar is packaged as a simple Docker container that can run an optional Ethereum node and index the Eigen Layer state, exposing the data via RPC.
+//! - **Real-Time Data Access**: Users can access real-time information on stakes, payment schedules, APRs, and more.
```
Comment (ypatil12) [inline sidecar/src/main.rs:39]: nit: payment -> reward

File: sidecar/src/reth_sidecar.rs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                    amount = ?claimed.claimedAmount,
+                    "Received RewardClaimed event"
+                );
+            }
+            _ => {
+                debug!("Received event not handled: {:?}", log);
+            }
+        }
+    }
+    let mut block = block.write().unwrap();
```
Comment (ypatil12) [inline sidecar/src/reth_sidecar.rs:64]: Won't we be blocking if there are read locks on the block, worst case taking more than 12 seconds to process the block ?

**Review by ypatil12 — COMMENTED** [review]
Minor comments

File: layr_rpc/src/lib.rs (hunk 2 above)
Comment (jac18281828) [reply layr_rpc/src/lib.rs:103]: Ah - k good feedback.    I'll fix that

File: sidecar/src/reth_sidecar.rs (hunk 5 above)
Comment (jac18281828) [reply sidecar/src/reth_sidecar.rs:64]: No, the read lock does not block.   The nice property of the read-write lock is the reads are always unlocked, while multiple writers would be blocked by each other.   In this case I don't anticipate multiple writers so it should never practically be blocked.

File: layr_rpc/src/lib.rs (hunk 1 above)
Comment (jac18281828) [reply layr_rpc/src/lib.rs:113]: This is the synchronized block of the sidecar versus the current block height.   I agree it's not super useful, we are still growing the project.   Let me share with you the api document on slack.

Comment (jac18281828) [reply layr_rpc/src/lib.rs:113]: > ie. `eth_blockNumber`. Do we also want to support JSON-RPC API with the `layr_rpc`?

I'm not following the question?

**Review by ypatil12 — APPROVED** [review]

**Pull request merged**